src/containers/Transaction/Transaction.js

```diff
@@ -0,0 +1,174 @@
+};
+
+const schema = yup.object().shape({
+  currencyName: yup
+    .string()
+    .required(),
+  amount: yup
+    .number()
+    .positive()
+    .integer()
+    .transform((val) => (Number.isNaN(val) ? undefined : val))
+    .required()
+});
+
+export const errorMessages = {
+  required: 'The field is required',
+  typeError: 'Value must be a number',
+  integer: 'Value must be an integer',
+  min: 'Amount must be a positive number'
+};
+
+const Transaction = ({ stockExchangeData }) => {
+  const {
+    reset,
+    control,
+    handleSubmit,
+    watch,
+    setValue,
+    getValues,
+    formState: { errors }
+  } = useForm({
+    defaultValues,
+    resolver: yupResolver(schema)
+  });
+  const [isBuying, currencyName, amount] = [watch('isBuying'), watch('currencyName'), watch('amount')];
+
+  const onSubmit = (data) => {
+    // eslint-disable-next-line no-console
+    console.log(data);
+  };
+
+  useEffect(() => {
+    setValue('price', stockExchangeData.find((dataItem) => dataItem.name === currencyName)?.price || '');
+  }, [currencyName, setValue, stockExchangeData]);
+
+  useEffect(() => {
+    if (amount && currencyName && !errors.amount) {
+      const total = stockExchangeData.find((dataItem) => dataItem.name === currencyName).price * getValues('amount');
+      setValue('total', total.toFixed(2));
+    }
+  }, [amount, currencyName, errors.amount, getValues, setValue, stockExchangeData]);
+
+  return (
+    <form onSubmit={handleSubmit(onSubmit)} className={styles.wrapper}>
+      <div className={styles.heading}>
+        <Text text="Transaction Type:" type="HEADING_5" />
+        <Text text={isBuying ? 'Buy' : 'Sell'} type="HEADING_5" state={isBuying ? 'SUCCESS' : 'ACCENT'} />
+        <Controller
+          name="isBuying"
+          control={control}
+          render={({ field }) => (
+            <Component.Switch
+              color="primary"
+              checked={field.value}
+              inputProps={{ 'aria-label': 'primary checkbox' }}
+              {...field}
+            />
+          )}
+        />
+      </div>
+      <div className={styles.form}>
+        <Component.FormControl variant="outlined" error={!!errors.currencyName}>
+          <Component.InputLabel id="currencyName">Name</Component.InputLabel>
+          <Controller
+            name="currencyName"
+            control={control}
+            render={({ field }) => (
+              <Component.Select
+                labelId="currencyName"
+                id="currencyName"
+                label="currencyName"
+                {...field}
+              >
+                {stockExchangeData.map((dataItem) => (
+                  <Component.MenuItem key={dataItem.uuid} value={dataItem.name}>
+                    {dataItem.name}
+                  </Component.MenuItem>
+                ))}
+              </Component.Select>
+            )}
+          />
+          <Component.FormHelperText>
+            {errorMessages[errors.currencyName?.type]}
```

Comment:
After apply my comment from above:
```suggestion
            {errors.currencyName?.message}
```